### PR TITLE
added route auth to plugin route config

### PR DESCRIPTION
**Pull request by trippo80**
Made it possible to set route authentication on a plugin having routes.

Seemed a bit too easy to implement so most likely I have forgotten something. Also - my first PR so be nice :)

### CODE REVIEW

**Comment by sirgallifrey** [comment]
You need to write some tests :)

**Comment by trippo80** [comment]
@sirgallifrey Done! (I hope :) )

File: lib/schema.js [hunk 1]
```diff
@@ -322,7 +322,8 @@ internals.register = Joi.object({
     once: Joi.boolean(),
     routes: Joi.object({
         prefix: Joi.string().regex(/^\/.+/),
-        vhost: internals.vhost
+        vhost: internals.vhost,
+        auth: Joi.string()
```
Comment (hueniverse) [inline lib/schema.js]: Why is this a string and not `internals.auth.allow(false)`?

File: lib/route.js [hunk 2]
```diff
@@ -78,6 +78,12 @@ exports = module.exports = internals.Route = function (route, connection, plugin
     this.settings.plugins = this.settings.plugins || {};            // Route-specific plugins settings, namespaced using plugin name
     this.settings.app = this.settings.app || {};                    // Route-specific application settings
 
+    // Apply plugin authentication configuration (superseded by route authentication configuration)
+
+    if (realm.modifiers.route.auth) {
```
Comment (hueniverse) [inline lib/route.js]: This can be `false`.

**Comment by hueniverse** [comment]
Someone wants to try and fix this with my comments before I close it?

**Comment by trippo80** [comment]
@hueniverse Hey - sorry for not responding earlier. Sadly I don't have time to look at this in the near future. Swamped at both work and private.

**Comment by hueniverse** [comment]
@trippo80 no worries. I'll close it for now. If the need comes up again, someone can pick it up and resubmit.

**Pull request closed without merging**